resources-generator/src/main/kotlin/dev/icerock/gradle/generator/js/JsStringsGenerator.kt

```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright 2022 IceRock MAG Inc. Use of this source code is governed by the Apache 2.0 license.
+ */
+
+package dev.icerock.gradle.generator.js
+
+import com.squareup.kotlinpoet.*
+import dev.icerock.gradle.generator.KeyType
+import dev.icerock.gradle.generator.LanguageType
+import dev.icerock.gradle.generator.StringsGenerator
+import dev.icerock.gradle.generator.js.JsMRGenerator.Companion.STRINGS_FALLBACK_FILE_URL_PROPERTY_NAME
+import dev.icerock.gradle.generator.js.JsMRGenerator.Companion.SUPPORTED_LOCALES_PROPERTY_NAME
+import dev.icerock.gradle.generator.js_jvm_common.generateFallbackAndSupportedLanguageProperties
+import kotlinx.serialization.json.buildJsonObject
+import kotlinx.serialization.json.put
+import org.gradle.api.file.FileTree
+import java.io.File
+
+class JsStringsGenerator(
+    stringsFileTree: FileTree,
+    mrClassPackage: String
+) : StringsGenerator(stringsFileTree) {
+
+    private val flattenClassPackage = mrClassPackage.replace(".", "")
+
+    override fun getClassModifiers(): Array<KModifier> = arrayOf(KModifier.ACTUAL)
+
+    override fun getPropertyModifiers(): Array<KModifier> = arrayOf(KModifier.ACTUAL)
+
+    override fun getPropertyInitializer(key: String) =
+        CodeBlock.of(
+            "StringResource(key = %S, supportedLocales = %N, fallbackFileUri = %N)",
+            key,
+            SUPPORTED_LOCALES_PROPERTY_NAME,
+            STRINGS_FALLBACK_FILE_URL_PROPERTY_NAME
+        )
+
+    override fun extendObjectBodyAtStart(classBuilder: TypeSpec.Builder) = Unit
+
+    override fun extendObjectBodyAtEnd(classBuilder: TypeSpec.Builder) = Unit
```

Comment:
instead of this you can add `, ObjectBodyExtendable by NOPObjectBodyExtendable()` to class declaration